Modules/Loadable/Segmentations/Widgets/qMRMLSegmentationGeometryWidget.cxx

```diff
@@ -0,0 +1,803 @@
+  this->GeometryImageData->GetSpacing(spacing);
+  double outputSpacing[3]  = {
+    spacing[this->SourceAxisIndexForInputAxis[0]],
+    spacing[this->SourceAxisIndexForInputAxis[1]],
+    spacing[this->SourceAxisIndexForInputAxis[2]] };
+
+  bool blocked = this->MRMLCoordinatesWidget_Spacing->blockSignals(true);
+  this->MRMLCoordinatesWidget_Spacing->setCoordinates(outputSpacing);
+  this->MRMLCoordinatesWidget_Spacing->blockSignals(blocked);
+
+  double origin[3] = { 0.0 };
+  this->GeometryImageData->GetOrigin(origin);
+  this->MRMLCoordinatesWidget_Origin->setCoordinates(origin);
+
+  vtkNew<vtkMatrix4x4> directions;
+  this->GeometryImageData->GetDirectionMatrix(directions);
+  for (int i=0; i<3; ++i)
+    {
+    for (int j=0; j<3; ++j)
+      {
+      this->MatrixWidget_Directions->setValue(i, j, directions->GetElement(i,j));
+      }
+    }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidgetPrivate::resetGeometryImageData()
+{
+  vtkNew<vtkMatrix4x4> identityMatrix;
+  identityMatrix->Identity();
+  this->GeometryImageData->SetImageToWorldMatrix(identityMatrix);
+  this->GeometryImageData->SetDimensions(0, 0, 0);
+}
+
+//-----------------------------------------------------------------------------
+bool qMRMLSegmentationGeometryWidgetPrivate::isSourceASegmentationWithBinaryLabelmapMaster()
+{
+  vtkMRMLTransformableNode* sourceNode = vtkMRMLTransformableNode::SafeDownCast(
+    this->MRMLNodeComboBox_SourceGeometryNode->currentNode() );
+  vtkMRMLSegmentationNode* sourceSegmentationNode = vtkMRMLSegmentationNode::SafeDownCast(sourceNode);
+  std::string binaryLabelmapName = vtkSegmentationConverter::GetBinaryLabelmapRepresentationName();
+  vtkSmartPointer<vtkOrientedImageData> sourceBinaryLabelmap;
+  if ( sourceSegmentationNode
+    && sourceSegmentationNode->GetSegmentation()->GetNumberOfSegments() > 0
+    && sourceSegmentationNode->GetSegmentation()->ContainsRepresentation(binaryLabelmapName)
+    && sourceSegmentationNode->GetSegmentation()->GetMasterRepresentationName() == binaryLabelmapName )
+    {
+    return true;
+    }
+
+  return false;
+}
+
+//-----------------------------------------------------------------------------
+bool qMRMLSegmentationGeometryWidgetPrivate::inputSegmentationCanBeResampled()
+{
+  if ( this->SegmentationNode != nullptr
+    &&  this->SegmentationNode->GetSegmentation()->GetNumberOfSegments() > 0
+    && this->SegmentationNode->GetSegmentation()->ContainsRepresentation(
+        vtkSegmentationConverter::GetBinaryLabelmapRepresentationName())
+    && this->SegmentationNode->GetSegmentation()->GetMasterRepresentationName()
+        == vtkSegmentationConverter::GetBinaryLabelmapRepresentationName() )
+    {
+    return true;
+    }
+  return false;
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidgetPrivate::matchInputAndSourceAxes(vtkMRMLNode* sourceNode)
```

Comment:
"match" does not indicate what this method actually does. "ComputeSourceAxisIndexForInputAxis" or something like that would be more informative.